hedera-node/src/main/java/com/hedera/services/store/tokens/BaseTokenStore.java

```diff
@@ -463,6 +463,27 @@ private ResponseCodeEnum tryAdjustment(AccountID aId, TokenID tId, long adjustme
 		return OK;
 	}
 
+	protected ResponseCodeEnum tryAdjustment(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
+		var relationshipSender = asTokenRel(senderAId, tId);
+		var relationshipReceiver = asTokenRel(receiverAId, tId);
+		if ((boolean) tokenRelsLedger.get(relationshipSender, IS_FROZEN) ||
+				(boolean) tokenRelsLedger.get(relationshipReceiver, IS_FROZEN)) {
+			return ACCOUNT_FROZEN_FOR_TOKEN;
+		}
+		if (!(boolean) tokenRelsLedger.get(relationshipSender, IS_KYC_GRANTED) ||
+				!(boolean) tokenRelsLedger.get(relationshipReceiver, IS_KYC_GRANTED)) {
+			return ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN;
+		}
+		long balanceSender = (long) tokenRelsLedger.get(relationshipSender, TOKEN_BALANCE);
+		if (balanceSender <= 0) {
+			return INSUFFICIENT_TOKEN_BALANCE;
+		}
+		tokenRelsLedger.set(relationshipSender, TOKEN_BALANCE, 0);
+		tokenRelsLedger.set(relationshipReceiver, TOKEN_BALANCE, 1);
```

Comment:
This would set the balance of ther sender to `0`. If he has 5 other NFTs we don't want to set his balance to 0. we want to reduce it by 1 and increase the receiver by 1